lib/vsc/utils/nagios.py

```diff
@@ -151,15 +150,13 @@ def __init__(self, nrange):
             @param nrange: nrange in [@][start:][end] format. If it is not a string, it is converted to
                           string and that string should allow conversion to float.
         """
-        self.log = getLogger(self.__class__.__name__, fname=False)
-
         if not isinstance(nrange, str):
             newnrange = str(nrange)
-            self.log.debug("nrange %s of type %s, converting to string (%s)", str(nrange), type(nrange), newnrange)
+            logging.debug("nrange %s of type %s, converting to string (%s)", str(nrange), type(nrange), newnrange)
             try:
                 float(newnrange)
             except ValueError:
-                self.log.raiseException("nrange %s (type %s) is not valid after conversion to string (newnrange %s)" %
+                logging.raiseException("nrange %s (type %s) is not valid after conversion to string (newnrange %s)" %
```

Comment:
this won't work, standard logging does not have raiseException. (with good reason). do the log and raising separately.

Reply:
I would suggest to use logging.exception here.

Reply:
if you really need it you can always make a small function log_and_raise. but typically, I would suggest  raise is adequate. do you really need the logging then?